gen/builders/builder_tipset.go

```diff
@@ -37,13 +37,14 @@ import (
 type TipsetVectorBuilder struct {
 	*BuilderCommon
 
-	// StagedMessages and tmpStateTracker are the staging area for messages.
-	StagedMessages  *Messages
-	tmpStateTracker *StateTracker
+	// StagedMessages is a staging area for messages.
+	StagedMessages *Messages
+	// StateTracker is used for staging messages, and it's scrapped and replaced
+	// by a fork at the PreRoot when committing applies.
+	StateTracker *StateTracker
```

Comment:
Simplified the temporary state tracker construction here. We now just have one state tracker, and we replace it in `CommitApplies` by recovering the pre-state.